File: fixtures/scale/kotlin/medium/src/calc/ast/AssignExpr.kt
package calc.ast

data class AssignExpr(
    val name: String,
    val value: Expr,
) : Expr {
    override fun describe(): String =
        name + " = " + value.describe()

    fun isSelfReference(): Boolean =
        value is VariableExpr && value.name == name
}

File: fixtures/scale/scala/medium/src/calc/ast/AssignExpr.scala
package calc.ast

final case class AssignExpr(name: String, value: Expr) extends Expr {

  override def describe: String = name + " = " + value.describe

  def isSelfReference: Boolean = value match {
    case VariableExpr(other) => other == name
    case _                   => false
  }
}
